bin/build.sh

```diff
@@ -58,112 +253,91 @@ if [ "${PORT_GIT}x" != "x" ]; then
-	if ! [ -f "${ca}" ]; then
-		echo "Certificate ${ca} required to clone from git" >&2
-		exit 4
-	fi
 	export GIT_SSL_CAINFO="${ca}"
 	deps="${PORT_GIT_DEPS}"
 fi
 
-fail=false
-for dep in $deps; do
-	if ! [ -f "${HOME}/zot/prod/${dep}/.env" ] && ! [ -f "${HOME}/zot/boot/${dep}/.env" ] && ! [ -f "/usr/zot/${dep}/.env" ] ; then
-		echo "Unable to find .env for dependency ${dep}" >&2
-		fail=true
-	fi
-done
-if $fail ; then
-	exit 4
-fi
+checkdeps ${deps}
 
-if ! [ -d "${PORT_ROOT}" ]; then
-	echo "PORT_ROOT ${PORT_ROOT} is not a directory" >&2
-	exit 8
-fi
-
-BASE_CFLAGS="-std=gnu11 -DNSIG=39 -D_XOPEN_SOURCE=600 -D_ALL_SOURCE -qascii -D_AE_BIMODAL=1 -D_ENHANCED_ASCII_EXT=0xFFFFFFFF -qfloat=ieee"
+BASE_CFLAGS="-DNSIG=39 -D_XOPEN_SOURCE=600 -D_ALL_SOURCE -qascii -D_AE_BIMODAL=1 -D_ENHANCED_ASCII_EXT=0xFFFFFFFF -qfloat=ieee"
+BASE_CXXFLAGS="-+ -DNSIG=39 -D_XOPEN_SOURCE=600 -D_ALL_SOURCE -qascii -D_AE_BIMODAL=1 -D_ENHANCED_ASCII_EXT=0xFFFFFFFF -qfloat=ieee"
 BASE_LDFLAGS="" 
 
 export CC=xlclang
-export CPP=xlclang++
+export CXX=xlclang++
 export CFLAGS="${BASE_CFLAGS} ${PORT_EXTRA_CFLAGS}"
+export CXXFLAGS="${BASE_CXXFLAGS} ${PORT_EXTRA_CXXFLAGS}"
 export LDFLAGS="${BASE_LDFLAGS} ${PORT_EXTRA_LDFLAGS}"
 
-for dep in $deps; do
-	if [ -f "${HOME}/zot/prod/${dep}/.env" ]; then
-		cd "${HOME}/zot/prod/${dep}"
-	elif [ -f "/usr/zot/${dep}/.env" ] ; then
-		cd "/usr/zot/${dep}"
-	elif [ -f "${HOME}/zot/boot/${dep}/.env" ]; then
-		cd "${HOME}/zot/boot/${dep}"
-	else 
-		echo "Internal error. Unable to find .env but earlier check should have caught this" >&2
-		exit 16
-	fi
-	. ./.env
-done
+setdepsenv $deps
 
 cd "${PORT_ROOT}" || exit 99
 
 if [ "${PORT_GIT}x" != "x" ]; then
-	if ! git --version >$STDOUT 2>$STDERR ; then
-		echo "git is required to download from the git repo" >&2
-		exit 4
-	fi
 	echo "Checking if git directory already cloned"
-	gitname=$(basename $PORT_GIT_URL)
-	dir=${gitname%%.*}
-	if [ -d "${dir}" ]; then
-		echo "Using existing git clone'd directory ${dir}"
-	else
-		echo "Clone and create ${dir}"
-		if ! git clone "${PORT_GIT_URL}" >$STDOUT 2>$STDERR; then
-                        echo "Unable to clone ${gitname} from ${PORT_GIT_URL}" >&2
-                        exit 4
-		else
-			chtag -R -h -tcISO8859-1 "${dir}"
-		fi
-	fi
+	dir=$( gitclone )
 fi	
 
 if [ "${PORT_TARBALL}x" != "x" ]; then
-	if ! curl --version >$STDOUT 2>$STDERR ; then
-		echo "curl is required to download a tarball" >&2
-		exit 4
-	fi
-	if ! gunzip --version >$STDOUT 2>$STDERR ; then
-		echo "gunzip is required to unzip a tarball" >&2
-		exit 4
-	fi
-	echo "Checking if tarball directory already created"
-	tarballz=$(basename $PORT_TARBALL_URL)
-	dir=${tarballz%%.tar.gz} 
-	if [ -d "${dir}" ]; then
-		echo "Using existing tarball directory ${dir}"
+	echo "Checking if tarball already downloaded"
+	dir=$( downloadtarball )
+fi	
+
+applypatches 
+
+cd "${PORT_ROOT}/${dir}" || exit 99
+
+# Proceed to build
+
+if [ "${PORT_GIT}x" != "x" ] && [ -f ./bootstrap ]; then
```

Comment:
What is `./bootstrap` here?

Reply:
If it's a 'git build' then many of them (but not all) require a boot strap step before configure which uses autoconf/automake to create the 'configure' file